plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java

```diff
@@ -1651,6 +1651,49 @@ public void testExternalLocationWithTrailingSpace()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testUnpartitionedTableExternalLocationOnTopOfTheBucket()
+    {
+        String topBucketName = "test-hive-unpartitioned-top-of-the-bucket-" + randomNameSuffix();
+        hiveMinioDataLake.getMinio().createBucket(topBucketName);
+        String tableName = "test_external_location_top_of_the_bucket_" + randomNameSuffix();
+
+        byte[] contents = "Trino\nSQL\non\neverything".getBytes(UTF_8);
+        hiveMinioDataLake.getMinioClient().putObject(topBucketName, contents, "data.txt");
+
+        assertUpdate(format(
+                "CREATE TABLE %s (" +
+                        "  a_varchar varchar) " +
+                        "WITH (" +
+                        "   external_location='%s'," +
+                        "   format='TEXTFILE')",
+                tableName,
+                format("s3://%s/", topBucketName)));
+        assertQuery("SELECT * FROM " + tableName, "VALUES 'Trino', 'SQL', 'on', 'everything'");
```

Comment:
This test fails because of 
https://github.com/trinodb/trino/blob/5295f3d5a66020f735e993911f743a5df860269f/lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HdfsFileSystem.java#L232-L236

The above mentioned code causes a file listing under:

`s3://test-hive-unpartitioned-top-of-the-bucket-yow77xmfak/f07d593c-3b0a-4267-ba49-4e3a405e2c78`

cc @alexjo2144

Reply:
>The above mentioned code causes a file listing under:


that's expected. the code there intentionally lists files from some random, non-existing location